### ISSUE TITLE
Running Gnome-related services can cause Neofetch to incorrectly detect WM as Gnome/mutter

### ISSUE THREAD

**Issue by josteink** [dylanaraps/neofetch#2298]
## Description

Running neofetch when using the sway window-manager will report Mutter (Gnome) if some Gnome-services are started.

It's not uncommon to also depend on a few gnome-related tools (Gnome keyring, Gnome Control Center, Gnome Polkit, dbus, whatever).

Activating any of those may result in a gnome-shell process being launched, making this resulting process-lookup return incorrect results, because both sway and gnome-shell is present. (This may affect other tiling window-managers like i3wm as well, but I haven't tested that).

https://github.com/dylanaraps/neofetch/blob/ccd5d9f52609bbdcd5d8fa78c4fdb0f12954125f/neofetch#L1909-L1938

This can be seen from a manual shell invocation:

````sh
$ echo $(ps -e | grep -m 1 -o -F \
                               -e arcan \
                               -e asc \
                               -e clayland \
                               -e dwc \
                               -e fireplace \
                               -e gnome-shell \
                               -e greenfield \
                               -e grefsen \
                               -e hikari \
                               -e kwin \
                               -e lipstick \
                               -e maynard \
                               -e mazecompositor \
                               -e motorcar \
                               -e orbital \
                               -e orbment \
                               -e perceptia \
                               -e river \
                               -e rustland \
                               -e sway \
                               -e ulubis \
                               -e velox \
                               -e wavy \
                               -e way-cooler \
                               -e wayfire \
                               -e wayhouse \
                               -e westeros \
                               -e westford \
                               -e weston)
gnome-shell
````

If removing the `-m 1` clause, we can see multiple processes being found, and I guess the problem is about (a lacking) handling of precedence:

````sh
$ echo $(ps -e | grep -o -F \
                               -e arcan \
                               -e asc \
                               -e clayland \
                               -e dwc \
                               -e fireplace \
                               -e gnome-shell \
                               -e greenfield \
                               -e grefsen \
                               -e hikari \
                               -e kwin \
                               -e lipstick \
                               -e maynard \
                               -e mazecompositor \
                               -e motorcar \
                               -e orbital \
                               -e orbment \
                               -e perceptia \
                               -e river \
                               -e rustland \
                               -e sway \
                               -e ulubis \
                               -e velox \
                               -e wavy \
                               -e way-cooler \
                               -e wayfire \
                               -e wayhouse \
                               -e westeros \
                               -e westford \
                               -e weston)
gnome-shell sway sway
````

- [X] Does this issue still occur in the master branch? Yes

#### Neofetch version

7.1.0

## Screenshot

![image](https://user-images.githubusercontent.com/411338/220305369-3faa70b6-aa74-4e55-a3f5-fa8d65313bad.png)

## Config file

None

## Verbose log

[neofetchlog.txt](https://github.com/dylanaraps/neofetch/files/10791997/neofetchlog.txt)

**Comment by hykilpikonna** [comment]
This is fixed in [HyFetch 1.4.7](https://github.com/hykilpikonna/hyfetch/releases/tag/1.4.7) by PR [#94](https://github.com/hykilpikonna/hyfetch/pull/94).

> This repo (dylanaraps/neofetch) seems no longer maintained.
> 
> [HyFetch](https://github.com/hykilpikonna/hyfetch) is a fork of neofetch with LGBTQ pride flags, but the repo also maintains an updated version of the original neofetch, addressing many pull requests that are not merged in the original repo.
> 
> Read the ["Running Updated Original Neofetch" section](https://github.com/hykilpikonna/hyfetch#running-updated-original-neofetch) for more info!

**Comment by josteink** [comment]
Thanks for the suggestion, @hykilpikonna , but I prefer my software to be about software, not politics 😄 

As for neofetch... If it's not gonna be fixed, its not gonna be fixed. Closing.
